### PR TITLE
Cache IvyNode#getdependencies

### PR DESCRIPTION
**Pull request by eed3si9n**
`IvyNode#getDependencies` is one of the hot spots frequently called during resolution. See sbt/ivy#4 for more analysis on `ResolveEngine.resolve`.

Adding caching at this level is not as invasive, and has a nice speedup. Using large-graph sample project, 17s resolution improved to 10s.

### CODE REVIEW

File: src/java/org/apache/ivy/core/resolve/IvyNode.java [hunk 1]
```diff
@@ -101,6 +101,8 @@
 
     private Collection loadedRootModuleConfs = new HashSet();
 
+    private Map loadedDependencies = new HashMap();
```
Comment (jsuereth) [inline src/java/org/apache/ivy/core/resolve/IvyNode.java]: You should denote what the types of the map are, via `/* <Types, Here> */` for now. I think ivy.next may be able to use generics, even.

Comment (jsuereth) [inline src/java/org/apache/ivy/core/resolve/IvyNode.java]: Also, when does this get cleared?

**Comment by jsuereth** [comment]
Generally, looks good.  When adding a cache, we need to worry about leakage, which is the question here.  I assume `IvyNode` isn't cached anywhere, besides when being used actively during resolution, so it should be ok.

In any case,  this is probably ok to merge, just wanted to check since I wasn't sure.

File: src/java/org/apache/ivy/core/resolve/IvyNode.java (hunk 1 above)
Comment (eed3si9n) [inline src/java/org/apache/ivy/core/resolve/IvyNode.java]: I'm assuming during the lifetime of a `IvyNode` its dependencies does not change.

Comment (jsuereth) [inline src/java/org/apache/ivy/core/resolve/IvyNode.java]: So `loadedDependencies` is other `IvyNode`s?  Maybe after this makes it through we'll check to see if we have any classloader leaky issues.  I'm not sure what the lifespan of `IvyNode` is, btu I think sbt erases it for sure.

Comment (eed3si9n) [inline src/java/org/apache/ivy/core/resolve/IvyNode.java]: Yea. `Collection/* <IvyNode> */`.

**Comment by eed3si9n** [comment]
Added the generics comment.

**Comment by jsuereth** [comment]
Cool, last change looks good.

**Pull request merged**